### ISSUE TITLE
Removing remove_empty_expression

### ISSUE THREAD

**Issue by YehorBoiar** [conjure-cp/conjure-oxide#427]
In the PR #370 @ozgurakgun pointed our that rule `remove_empty_expression` doesn't make much sence becuase the same thing should happen in a higher priority rule `apply_eval_constant`.

Should we just get rid of that rule?

**Comment by niklasdewally** [comment]
There are a few rules that could just be folded into the total/partial evaluators, so I agree with this.

Once #401 is merged, you should be able to find these quite easily.

**Comment by YehorBoiar** [comment]
I checked #401. Now, new `choose_rewrite` would warn us about rules that are applicable, but not being applied as they are of a low priority.

So, I can just go through all warn messages that we are getting from that function right now. See what rules are never being applied. Open a PR where we would discuss removing those kind of rules. @ozgurakgun @niklasdewally what do you think?

**Comment by niklasdewally** [comment]
> I checked #401. Now, new `choose_rewrite` would warn us about rules that are applicable, but not being applied as they are of a low priority.
> 
> So, I can just go through all warn messages that we are getting from that function right now. See what rules are never being applied. Open a PR where we would discuss removing those kind of rules. @ozgurakgun @niklasdewally what do you think?

Perfect plan!

**Comment by ozgurakgun** [comment]
sounds good to me too
